### PR TITLE
Feat(eos_cli_config_gen): Extend router_isis data-model

### LINKED ISSUE #3390: Extend isis data model
### Enhancement summary

Add support in the router_isis data model for:

`spf-interval <values>`
`graceful-restart`
`bfd all-interfaces` under address-family

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Allow the above mentioned options to be set within a data model

### Describe the solution you would like

eos_cli_config_gen updated to support these options.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

Add support in the router_isis data model for:

spf-interval <values>
graceful-restart
bfd all-interfaces under address-family

## Related Issue(s)

Fixes #3390 

## Component(s) name

`arista.avd.eos_cli_config_gen`

### User Checklist

- [x] Verify command ordering in EOS CLI

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-isis.j2 [hunk 1]
```diff
@@ -62,6 +62,32 @@ router isis {{ router_isis.instance }}
...
+{%         if router_isis.graceful_restart.enabled is arista.avd.defined(true) %}
+   graceful-restart
+{%         endif %}
+{%         if router_isis.graceful_restart.restart_hold_time is arista.avd.defined %}
+   graceful-restart restart-hold-time {{ router_isis.graceful_restart.restart_hold_time }}
+{%         endif %}
+{%         if router_isis.graceful_restart.t2.wait_time is arista.avd.defined %}
+   graceful-restart t2 level-1 {{ router_isis.graceful_restart.t2.wait_time }}
+   graceful-restart t2 level-2 {{ router_isis.graceful_restart.t2.wait_time }}
+{%         else %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-isis.j2]: Please remove the wait_time option. Even though the CLI includes this abstraction, we should not replicate it here. Keep the level-1 and level-2

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-isis.j2 [hunk 2]
```diff
@@ -62,6 +62,32 @@ router isis {{ router_isis.instance }}
...
+{%             if router_isis.graceful_restart.t2.level_1_wait_time is arista.avd.defined %}
+   graceful-restart t2 level-1 {{ router_isis.graceful_restart.t2.level_1_wait_time }}
+{%             endif %}
+{%             if router_isis.graceful_restart.t2.level_2_wait_time is arista.avd.defined %}
+   graceful-restart t2 level-1 {{ router_isis.graceful_restart.t2.level_2_wait_time }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-isis.j2]:
```suggestion
   graceful-restart t2 level-2 {{ router_isis.graceful_restart.t2.level_2_wait_time }}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-isis.j2 [hunk 3]
```diff
@@ -108,6 +137,9 @@ router isis {{ router_isis.instance }}
 {%         if router_isis.address_family_ipv6.maximum_paths is arista.avd.defined %}
       maximum-paths {{ router_isis.address_family_ipv6.maximum_paths }}
 {%         endif %}
+{%         if router_isis.address_family_ipv4.bfd_all_interfaces is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-isis.j2]:
```suggestion
{%         if router_isis.address_family_ipv6.bfd_all_interfaces is arista.avd.defined(true) %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml [hunk 4]
```diff
@@ -194,3 +200,56 @@ keys:
...
+                min: 5
+                max: 300
+                description: LSP database sync wait time in seconds
+              level_1_wait_time:
+                type: int
+                convert_types:
+                  - str
+                min: 5
+                max: 300
+                description: Number of seconds
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml]:
```suggestion
                description: Level-1 LSP database sync wait time in seconds.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml [hunk 5]
```diff
@@ -194,3 +200,56 @@ keys:
...
+              level_2_wait_time:
+                type: int
+                convert_types:
+                  - str
+                min: 5
+                max: 300
+                description: Number of seconds
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml]:
```suggestion
                description: Level-2 LSP database sync wait time in seconds.
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-isis.j2 (hunk 1 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-isis.j2]: Fixed!

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-isis-new.cfg [hunk 6]
```diff
@@ -78,15 +78,22 @@ router isis EVPN_UNDERLAY
    mpls ldp sync default
    timers local-convergence-delay 15000 protected-prefixes
    advertise passive-only
+   spf-interval 250 10
+   graceful-restart
+   graceful-restart restart-hold-time 10
+   graceful-restart t2 level-1 10
+   graceful-restart t2 level-2 20
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-isis-new.cfg]: Ordering is different on my device (cEOS 4.31)

```
router isis BLAH
   spf-interval 250 10
   graceful-restart
   graceful-restart t2 level-1 10
   graceful-restart t2 level-2 20
   graceful-restart restart-hold-time 10
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml [hunk 7]
```diff
@@ -112,6 +112,9 @@ keys:
               - "str"
             min: 1
             max: 128
+          bfd_all_interfaces:
+            type: bool
+            description: Enable BFD on all interfaces
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml]:
```suggestion
            description: Enable BFD on all interfaces.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml [hunk 8]
```diff
@@ -152,6 +155,9 @@ keys:
...
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml]:
```suggestion
            description: Enable BFD on all interfaces.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml [hunk 9]
```diff
@@ -194,3 +200,49 @@ keys:
...
+        keys:
+          enabled:
+            type: bool
+          restart_hold_time:
+            type: int
+            convert_types:
+              - str
+            min: 5
+            max: 300
+            description: Number of seconds
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml]:
```suggestion
            description: Number of seconds.
```

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml]: and other descriptions

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-isis-new.cfg (hunk 6 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-isis-new.cfg]: Fixed!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml (hunk 9 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_isis.schema.yml]: Please add dots to every description you are adding in this file

**Review by gmuloc — APPROVED** [review]
LGTM - thanks!
tested new generated config in CLI in cEOS 4.31

**Pull request merged**